packages/browser/src/plugins/in-app-plugin/index.ts

```diff
@@ -2,208 +2,280 @@ import { Analytics } from '../../core/analytics'
+        void _analytics.track(JourneysEvents.Metric, {
+          deliveryId: deliveryId,
+          metric: JourneysEvents.Opened,
+        })
+        return
+      }
+      const broadcastId: Number =
+        message?.properties?.gist?.broadcast?.broadcastIdInt
+      if (broadcastId) {
+        const templateId = message?.properties?.gist?.broadcast?.templateId
+        void _analytics.track(JourneysEvents.Content, {
+          actionType: JourneysEvents.ViewedContent,
+          contentId: broadcastId,
+          templateId: templateId,
+          contentType: ContentType,
+        })
+      }
+    })
+
+    _gist.events.on('messageAction', (params: any) => {
+      const deliveryId: string = params?.message?.properties?.gist?.campaignId
+      if (settings.events) {
+        _eventTarget.dispatchEvent(
+          newEvent(InAppEvents.MessageAction, {
+            messageId: params.message.messageId,
+            deliveryId: deliveryId,
+            action: params.action,
+            name: params.name,
+            actionName: params.name,
+            actionValue: params.action,
+            message: {
+              dismiss: function () {
+                _gist.dismissMessage(params.message.instanceId)
+              },
+            },
+          })
+        )
+      }
+      if (params.action == 'gist://close') {
+        return
+      }
+      if (typeof deliveryId != 'undefined' && deliveryId != '') {
+        void _analytics.track(JourneysEvents.Metric, {
+          deliveryId: deliveryId,
+          metric: JourneysEvents.Clicked,
+          actionName: params.name,
+          actionValue: params.action,
+        })
+        return
+      }
+      const broadcastId: Number =
+        params?.message?.properties?.gist?.broadcast?.broadcastIdInt
+      if (broadcastId) {
+        const templateId: Number =
+          params?.message?.properties?.gist?.broadcast?.templateId
+        void _analytics.track(JourneysEvents.Content, {
+          actionType: JourneysEvents.ClickedContent,
+          contentId: broadcastId,
+          templateId: templateId,
+          contentType: ContentType,
+          actionName: params.name,
+          actionValue: params.action,
+        })
+      }
+    })
 
-        const page:string = ctx.event?.properties?.name ?? ctx.event?.properties?.url;
-        if(typeof page === 'string' && page.length > 0) {
-            Gist.setCurrentRoute(page);
+    _gist.events.on('eventDispatched', (gistEvent: any) => {
+      if (gistEvent.name == 'analytics:track') {
+        const trackEventName: string = gistEvent.payload?.event
+        if (typeof trackEventName === 'undefined' || trackEventName == '') {
+          return
         }
+        void _analytics.track(
+          trackEventName,
+          gistEvent.payload?.properties,
+          gistEvent.payload?.options
+        )
+      }
+    })
+  }
 
-        return ctx;
-    }
+  function page(ctx: Context): Promise<Context> {
```

Comment:
async?

Reply:
We're not using any `await` in the function so its not needed.